django/views/i18n.py

```diff
@@ -272,18 +234,20 @@ def get_javascript_catalog(locale, domain, packages):
     pdict = {}
     maxcnts = {}
     catalog = {}
-    for k, v in t.items():
-        if k == '':
+    trans_fallback_cat = trans._fallback._catalog if trans._fallback else {}
+    for key in itertools.chain(trans_cat.keys(), trans_fallback_cat.keys()):
+        if key == '' or key in catalog:
             continue
-        if isinstance(k, six.string_types):
-            catalog[k] = v
-        elif isinstance(k, tuple):
-            msgid = k[0]
-            cnt = k[1]
+        value = trans_cat.get(key, trans_fallback_cat.get(key, ''))
```

Comment:
If that's the case, wouldnt the code a bit better if you iterate it in a different way so that you don't have to perform two dict lookups (that will always have a value for the first half and always fail the first get and have a value for the second half)?

One very simple change would be to change .keys() with .items() and iterate like:
`for key, value in itertools.chain(trans_cat.items(), trans_fallback_cat.items()):`
With this you don't need to do `value = trans_cat.get(key, trans_fallback_cat.get(key, ''))`